### PR TITLE
Update tests for globalization cleanup

### PR DESCRIPTION
**Pull request by hughbe**
/cc @tarekgh @stephentoub @jkotas

### CODE REVIEW

File: src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoCurrencyDecimalDigits.cs [hunk 1]
```diff
@@ -19,28 +18,35 @@ public static IEnumerable<object[]> CurrencyDecimalDigits_TestData()
...
+        {
+            var format = new NumberFormatInfo();
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("value", "CurrencyDecimalDigits", () => format.CurrencyDecimalDigits = value);
+        }
+
         [Fact]
-        public void CurrencyDecimalDigits_Set_Invalid()
+        public void CurrencyDecimalDigits_SetReadOnly_ThrowsInvalidOperationException()
         {
-            AssertExtensions.Throws<ArgumentOutOfRangeException>("CurrencyDecimalDigits", () => new NumberFormatInfo().CurrencyDecimalDigits = -1);
```
Comment (tarekgh) [inline src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoCurrencyDecimalDigits.cs]: >Throws<ArgumentOutOfRangeException>("CurrencyDecimalDigits", () => new NumberFormatInfo().CurrencyD [](start = 29, length = 99)

why did you remove this?

**Comment by karelz** [comment]
@krwq @tarekgh can you please help with the test failures?

**Comment by tarekgh** [comment]
test corefx-ci (macOS x64_Debug)

**Comment by tarekgh** [comment]
Looks we still waiting to get the updates from coreclr side.

**Comment by hughbe** [comment]
I don't think we've merged that PR yet!

**Comment by tarekgh** [comment]
https://github.com/dotnet/coreclr/pull/22591

**Comment by tarekgh** [comment]
I restarted the CI on the coreclr PR hopefully we get a clean result. thanks @hughbe

**Comment by stephentoub** [comment]
Cherry-picked into https://github.com/dotnet/corefx/pull/35891
